test/controllers/application_helper_test.rb

```diff
@@ -21,4 +21,39 @@ def test_add_args_to_url
     assert_equal("/blah\x80",
                  add_args_to_url("/blah\x80", x: "foo\xA0"))
   end
+
+  def test_textile_markup_should_be_escaped
+  	textile = "**Bold**"
+  	escaped = "&lt;div class=&quot;textile&quot;&gt;&lt;p&gt;&lt;b&gt;Bold&lt;/b&gt;&lt;/p&gt;&lt;/div&gt;"
+		assert_equal escaped, escape_html(textile.tpl)
+  end
+
+  def test_conditional_content_tags
+    assert_nil(content_tag_if(nil, :p, "Hello world!"))
```

Comment:
I don't really like sets of tests like this since the first one to fail stops all the other ones to not get run which may reveal more.  The general recommendation I've heard is one assert per test.  That's not currently true in our code base by a long shot since I didn't know that when the project got started, but I've been moving more an more in that direction.